src/typ.rs

```diff
@@ -2205,6 +2216,15 @@ fn infer_let(
     let pattern = PatternTyper::new(env, level).unify(pattern, value_typ)?;
     let then = infer(then, level, env)?;
     let typ = then.typ();
+
+    // Check that any type annotation is accurate.
+    if let Some(ann) = annotation {
+        let ann_typ = env
+            .type_from_ast(ann, &mut hashmap![], NewTypeAction::MakeGeneric)
```

Comment:
Is it okay for me to pass in hashmap![] as the vector of type variables? I assume so, based on my understanding of what this function is doing, but I'm a little fuzzy on some of the details

Reply:
Oops I accidentally resolved the conversation and I don't know how to undo it on mobile.

It's a hashmap rather than a vector, but you are using it correctly, yes. It's how we keep track of what type variables have what names if we are building multiple types at once (i.e. function arguments)

Reply:
Ah yeah, I meant hashmap. I was mostly wondering whether you could think of any cases where I would need to collect type variables when handling let annotations.

It sounds like the answer is no though, thanks!